### PR TITLE
feat(signet page): #MED-256 create signet page

### PR DESCRIPTION
**Pull request by niko-amosse**
## Describe your changes

create Signet Page and redirect sidebar to this page

image

## Checklist tests

check the redirection and the display

## Issue ticket number and link

[MED-256](https://jira.support-ent.fr/browse/MED-256)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/filter-layout/FilterLayout.tsx [hunk 1]
```diff
@@ -91,7 +101,21 @@ export const FilterLayout: React.FC<FilterLayoutProps> = ({
...
+            t("mediacentre.signets.themes.orientation"),
+          );
+          return (
+            (type === t("mediacentre.signets.themes.orientation") &&
+              containOrientation) ||
+            (type === t("mediacentre.signets.without.theme") &&
+              !containOrientation)
+          );
+        });
+      return matchesDiscipline || matchesLevel || matchesType || matchesTheme;
```
Comment (niko-amosse) [inline frontend/src/components/filter-layout/FilterLayout.tsx:118]: ajout du match themes (toute la gestion des filtres sera revu dans un autre ticket)

File: frontend/src/components/sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -27,9 +25,6 @@ export const Sidebar: React.FC<SidebarProps> = ({
   const { t } = useTranslation();
   const sidebarRef = useRef<HTMLDivElement>(null);
 
-  const { data: actions } = useActions();
-  const canAccessSignet = isActionAvailable("signets", actions);
-
```
Comment (niko-amosse) [inline frontend/src/components/sidebar/Sidebar.tsx:32]: plus besoin des droits des signets dans la sidebar, ca sera utilisé directement sur la page des signets pour faire varier le contenu en fonction des droits

File: frontend/src/components/sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -74,16 +69,12 @@ export const Sidebar: React.FC<SidebarProps> = ({
...
-              selected={location.pathname === "/signet"}
-            />
-          </a>
-        )}
+        <SidebarIcon
+          action={() => navigate("/signets")}
+          icon={<BookmarkIcon />}
+          name={`${t("mediacentre.sidebar.signets")}`}
+          selected={location.pathname === "/signets"}
+        />
```
Comment (niko-amosse) [inline frontend/src/components/sidebar/Sidebar.tsx:77]: finito le angular, place au react

File: frontend/src/routes/signets/index.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+import "~/styles/page/search.scss";
+
+export const SignetPage: React.FC = () => {
+  const { t } = useTranslation();
+  const { refetchPins } = usePinProvider();
+  const { alertType, alertText, setAlertText } = useAlertProvider();
+
+  // const { data: actions } = useActions(); // POUR LA SUITE
+  // const canAccessSignet = isActionAvailable("signets", actions); // POUR LA SUITE
```
Comment (niko-amosse) [inline frontend/src/routes/signets/index.tsx:26]: les fameux droits qui seront utilisés plus tard

File: frontend/src/components/filter-layout/FilterLayout.tsx [hunk 5]
```diff
@@ -91,7 +101,21 @@ export const FilterLayout: React.FC<FilterLayoutProps> = ({
         selectedCheckboxesTypes.some((type) =>
           resource.document_types.includes(type),
         );
-      return matchesDiscipline || matchesLevel || matchesType;
+      const matchesTheme =
+        resource?.document_types &&
+        selectedCheckboxesThemes.length > 0 &&
```
Comment (ThomasSevagen) [inline frontend/src/components/filter-layout/FilterLayout.tsx:106]: `selectedCheckboxesThemes.length` ou `!!selectedCheckboxesThemes.length`

**Review by ThomasSevagen — APPROVED** [review]

**Review by Mattox0 — APPROVED** [review]
Les filtres ne sont pas bons, mais ils seront changés dans un autre ticket

**Pull request merged**